Sources/ScheduledTasks.php

```diff
@@ -458,6 +458,17 @@ function scheduled_daily_maintenance()
 			'oldLogins' => time() - (!empty($modSettings['loginHistoryDays']) ? 60 * 60 * 24 * $modSettings['loginHistoryDays'] : 2592000),
 	));
 
+	// Run Cache housekeeping
+	if (!empty($image_proxy_enabled))
+	{
+		global $proxyhousekeeping;
```

Comment:
^ Can we add this as a constructor argument to ProxyServer which stores the value as a property.

Reply:
The issue is here that this logic:
https://github.com/albertlast/SMF2.1/blob/5340d887d2a01da781ebb86da83a1359f0975476/proxy.php#L232-L236
run in this moment when you load the file with require_once,
wihtout loading the file i can't set the constructor.